pkg/prometheus/server/operator.go

```diff
@@ -1518,13 +1518,8 @@ func (c *Operator) createOrUpdateConfigurationSecret(ctx context.Context, p *mon
 	}
 
 	if p.Spec.Alerting != nil {
-		for i, am := range p.Spec.Alerting.Alertmanagers {
-			if err := store.AddBasicAuth(ctx, p.GetNamespace(), am.BasicAuth, fmt.Sprintf("alertmanager/auth/%d", i)); err != nil {
```

Comment:
I don't get the reason why we move this logic from this place to the `pkg/prometheus/store.go`

Reply:
I noticed some similar functions in the `pkg/prometheus/store.go`, such as `AddRemoteWritesToStore`, so I moved it to this file to standardize it. 😄

Reply:
Looks good to me :)

Indeed a lot of asset store code is already present in the store package